pkg/profile/chart_builder.go

```diff
@@ -0,0 +1,116 @@
+package profile
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+	"strings"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ChartBuilder will build helm chart resources.
+type ChartBuilder struct {
+	BuilderConfig
+}
+
+// Build a single artifact from a profile artifact and installation.
+func (c *ChartBuilder) Build(artifact profilesv1.Artifact, installation profilesv1.ProfileInstallation, definition profilesv1.ProfileDefinition) ([]Artifact, error) {
+	a := Artifact{Name: artifact.Name}
+	helmRelease := c.makeHelmRelease(artifact, installation, definition.Name)
+	a.Objects = append(a.Objects, helmRelease)
+	if artifact.Chart.Path != "" {
+		if c.GitRepositoryNamespace == "" && c.GitRepositoryName == "" {
+			return nil, fmt.Errorf("in case of local resources, the flux gitrepository object's details must be provided")
+		}
+		helmRelease.Spec.Chart.Spec.Chart = filepath.Join(c.RootDir, "artifacts", artifact.Name, artifact.Chart.Path)
+		branch := installation.Spec.Source.Branch
+		if installation.Spec.Source.Tag != "" {
+			branch = installation.Spec.Source.Tag
+		}
+		a.RepoURL = installation.Spec.Source.URL
+		a.SparseFolder = definition.Name
+		a.Branch = branch
+		a.PathsToCopy = append(a.PathsToCopy, artifact.Chart.Path)
+	}
+	if artifact.Chart.URL != "" {
```

Comment:
should this be an else?

Reply:
I don't know. :D It's a full copy of the original builder code. :D I'll take a closer look. :)

Reply:
> // This is an optional value. If defined, it takes precedence over other Chart fields.
This is the Path. So I think, it doesn't matter?